docs/src/modules/utils/parseTest.js

```diff
@@ -0,0 +1,92 @@
+import * as babel from '@babel/core';
+import { readFile } from 'fs-extra';
+import * as path from 'path';
+
+const workspaceRoot = path.join(__dirname, '../../../../');
+const babelConfigPath = path.join(workspaceRoot, 'babel.config.js');
+
+function withExtension(filepath, extension) {
+  return path.join(
+    path.dirname(filepath),
+    path.basename(filepath, path.extname(filepath)) + extension,
+  );
+}
+
+/**
+ * @param {string} filename
+ * @param {string} configFilePath
+ */
+async function parseWithConfig(filename, configFilePath) {
+  const source = await readFile(filename, { encoding: 'utf8' });
+  const partialConfig = babel.loadPartialConfig({
+    configFile: configFilePath,
+    filename,
+  });
+  return babel.parseAsync(source, partialConfig.options);
+}
+
+function findConformanceDescriptor(program) {
+  const { types: t } = babel;
+
+  let descriptor = {};
+  babel.traverse(program, {
+    CallExpression(babelPath) {
+      const { node: callExpression } = babelPath;
+      const { callee } = callExpression;
+      if (t.isIdentifier(callee) && callee.name === 'describeConformance') {
+        // describeConformance(element, () => options);
+        descriptor = callExpression.arguments[1].body;
+      }
+    },
+  });
+
+  if (descriptor.type != null && !t.isObjectExpression(descriptor)) {
+    throw new Error(`Expected an object expression as a descriptor but found ${descriptor.type}`);
+  }
+
+  return descriptor;
+}
+
+// extracts the property of window.*
+function getRefInstance(valueNode) {
+  if (!babel.types.isMemberExpression(valueNode)) {
+    throw new Error('Expected window.*Element in refInstanceOf');
+  }
+
+  return valueNode.property.name;
+}
+
+/**
+ * @typedef {Object} ParseResult
+ * @property {string?} forwardsRefTo
+ */
+
+/**
+ *
+ * @param {string} componentFilename
+ * @returns {ParseResult}
+ */
+export default async function parseTest(componentFilename) {
+  const testFilename = withExtension(componentFilename, '.test.js');
+  const babelParseResult = await parseWithConfig(testFilename, babelConfigPath);
+  const descriptor = findConformanceDescriptor(babelParseResult.program);
+
+  const result = {
+    forwardsRefTo: undefined,
+  };
+
+  const { properties = [] } = descriptor;
+  properties.forEach(property => {
+    const key = property.key.name;
+
+    switch (key) {
+      case 'refInstanceof':
+        result.forwardsRefTo = getRefInstance(property.value);
```

Comment:
Why is a RegEx not enough?

Reply:
It does not cover the full syntax of JS (regular expressions can't cover context free languages). We had plenty of issues with those regexes in the past which are all covered by using the AST. I also don't want to be bothered by order of the keys, holes etc.

This functionality will be extended (getting rid of `@inheritComponent`), documenting spread etc.

Reply:
I don't see the need in this very use case, but I agree that an AST has more potential. Let's see if we can live up to it.

Reply:
Well no but it's part of a larger effort and is meant to parse to whole descriptor not just `refInstanceOf: window.*`.